File: example/src/main/java/com/vmloft/develop/library/example/demo/scheme/SchemeActivity.java
package com.vmloft.develop.library.example.demo.scheme;

import android.net.Uri;

import com.vmloft.develop.library.example.R;
import com.vmloft.develop.library.example.common.AppActivity;
import com.vmloft.develop.library.tools.utils.VMLog;

import java.util.List;

/**
 * Create by lzan13 on 2020-01-02 20:46
 * 测试 Scheme 方式打开 app
 */
public class SchemeActivity extends AppActivity {
    @Override
    protected int layoutId() {
        return R.layout.activity_scheme;
    }

    @Override
    protected void initUI() {
        super.initUI();
    }

    @Override
    protected void initData() {
        Uri uri = getIntent().getData();
        if (uri != null) {
            // 完整的url信息
            String url = uri.toString();
            VMLog.d("url: " + uri);
            // scheme部分
            String scheme = uri.getScheme();
            VMLog.d("scheme: " + scheme);
            // host部分
            String host = uri.getHost();
            VMLog.d("host: " + host);
            //port部分
            int port = uri.getPort();
            VMLog.d("port: " + port);
            // 访问路劲
            String path = uri.getPath();
            VMLog.d("path: " + path);
            // 参数
            List<String> pathSegments = uri.getPathSegments();
            VMLog.d("pathSegments: " + pathSegments.toString());
            // Query部分
            String query = uri.getQuery();
            VMLog.d("query: " + query);
            //获取指定参数值
            String goodsId = uri.getQueryParameter("id");
            VMLog.d("id: " + goodsId);
        }
    }
}

File: example/src/main/java/com/vmloft/develop/library/example/demo/scheme/SchemeActivity.kt
package com.vmloft.develop.library.example.demo.scheme

import com.alibaba.android.arouter.facade.annotation.Route
import com.vmloft.develop.library.example.R.layout
import com.vmloft.develop.library.example.base.BaseActivity
import com.vmloft.develop.library.tools.utils.logger.VMLog

/**
 * Create by lzan13 on 2020-01-02 20:46
 * 测试 Scheme 方式打开 app
 */
@Route(path = "/VMLoft/Scheme")
class SchemeActivity : BaseActivity() {

    override fun layoutId(): Int = layout.activity_scheme
    override fun initUI() {
        super.initUI()
        setTopTitle("Url 跳转")
    }

    override fun initData() {
        val uri = intent.data
        if (uri != null) {
            // 完整的url信息
            val url = uri.toString()
            VMLog.d("url: $uri")
            // scheme部分
            val scheme = uri.scheme
            VMLog.d("scheme: $scheme")
            // host部分
            val host = uri.host
            VMLog.d("host: $host")
            //port部分
            val port = uri.port
            VMLog.d("port: $port")
            // 访问路劲
            val path = uri.path
            VMLog.d("path: $path")
            // 参数
            val pathSegments = uri.pathSegments
            VMLog.d("pathSegments: $pathSegments")
            // Query部分
            val query = uri.query
            VMLog.d("query: $query")
            //获取指定参数值
            val goodsId = uri.getQueryParameter("id")
            VMLog.d("id: $goodsId")
        }
    }
}